substrate-node/support/src/resources.rs

```diff
@@ -175,4 +195,39 @@ mod test {
         let su = resources.get_su();
         assert_eq!(su, 3);
     }
+
+    #[test]
+    fn test_resources_diff() {
+        let resources = Resources {
+            hru: 4 * GIGABYTE as u64 * 1024,
+            cru: 64,
+            mru: 64 * GIGABYTE as u64,
+            sru: 0,
+        };
+
+        let new_resources = Resources {
+            hru: 4 * GIGABYTE as u64 * 1024,
+            cru: 64,
+            mru: 64 * GIGABYTE as u64,
+            sru: 0,
+        };
+
+        assert_ok!(new_resources.has_changed(resources, 1));
+
+        let resources = Resources {
+            hru: 4 * GIGABYTE as u64 * 1024,
+            cru: 64,
+            mru: 64 * GIGABYTE as u64,
+            sru: 0,
+        };
+
+        let new_resources = Resources {
+            hru: 4 * GIGABYTE as u64 * 1024,
+            cru: 64,
+            mru: 40 * GIGABYTE as u64,
+            sru: 0,
+        };
+
+        assert_err!(new_resources.has_changed(resources, 1), ());
```

Comment:
Maybe it should be tested on critical scenarios
examples:
- 1% tolerance on 2 config with 1000 GB and 989 GB to evaluate sensitivity / rounding behavior close to the targeted threshold value (990 GB)